Added version to the top of the aa model

carlosvdr opened the pull request:
Issue: https://github.com/semiotic-ai/autoagora/issues/42

carlosvdr committed:
feat: Added version to the top of the aa model

Signed-off-by: carlos.vdr <carlos.vdr@semiotic.ai>

aasseman reviewed (changes requested):
- This is not the correct way of checking the version. One of the reasons is that `poetry` is a dev dependency, and is not installed in the container (see the Dockerfile), nor is it required by the pip package. In general, it is unnecessarily risky to run/depend on external binaries.
  See https://docs.python.org/3/library/importlib.metadata.html#distribution-versions
- Smaller note, can we have something like `Generated by AutoAgora {version}`?

aasseman commented on autoagora/model_builder.py:
```diff
@@ -15,7 +16,8 @@
     logs_db = LogsDB(pgpool)
     agora_entries = []
-
+    aa_version = version("wheel")
```
```suggestion
    aa_version = version("autoagora")
```

aasseman commented:
Looks good. Could you write a unit test to make sure that it works as expected?

carlosvdr committed:
test: Added a test to ensure AutoAgora version is added

Signed-off-by: carlos.vdr <carlos.vdr@semiotic.ai>

aasseman commented on tests/test_model_builder.py:
```diff
@@ -0,0 +1,50 @@
+            pattern = r"\n# Generated by AutoAgora \d+\.\d+\.\d+"
+            # To ensure a version is being obtained
+            assert re.match(pattern, model), f"{model} does not match pattern {pattern}"
+            assert model.startswith("\n# Generated by AutoAgora")
```
Isn't that line redundant with the one above?

carlosvdr replied:
Was thinking on the possibility of the version not showing in that specific way, but actually you are right if it doesnt show in that way it is definetly wrong, let me remove it

aasseman reviewed (changes requested):
Looks great! 
Could you remove the newline before `Generated by...`. It just adds an empty newline at the very top of the Agora document.

aasseman commented:
Perfect! I'm gonna do a bit of the old test in prod and then we're good to go!

aasseman commented:
Need to merge #63 first

aasseman commented:
Please rebase again onto latest master, then I'll test again for good measure

aasseman committed:
build: Remove .git from .gcloudignore

We need .git during the build process for poetry-dynamic-versioning to
work properly.

Signed-off-by: Alexis Asseman <alexis@semiotic.ai>

aasseman reviewed (approved):
LGTM, tested in prod.
